File: LeetCodeCPP/283.move-zeroes.cpp
/*
 * @lc app=leetcode id=283 lang=cpp
 *
 * [283] Move Zeroes
 */
#include <vector>

using namespace std;

// @lc code=start
class Solution
{
public:
    void moveZeroes(vector<int> &nums)
    {
        int lastNoZeroIdx = 0;
        for (size_t i = 0; i < nums.size(); i++)
        {
            if (nums[i] != 0)
                nums[lastNoZeroIdx++] = nums[i];
        }
        for (size_t i = lastNoZeroIdx; i < nums.size(); i++)
        {
            nums[i] = 0;
        }
    }
};
// @lc code=end

File: LeetCodePython/283.move-zeroes.py
#
# @lc app=leetcode id=283 lang=python3
#
# [283] Move Zeroes
#

from typing import List


# @lc code=start
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        last_non_zero_index = 0
        for i in range(0, len(nums)):
            if nums[i] != 0:
                nums[last_non_zero_index] = nums[i]
                last_non_zero_index += 1

        for i in range(last_non_zero_index, len(nums)):
            nums[i] = 0


# @lc code=end
